exist-core/src/main/java/org/exist/util/io/ByteArrayConsumer.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * eXist Open Source Native XML Database
+ * Copyright (C) 2001-2019 The eXist Project
+ * http://exist-db.org
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package org.exist.util.io;
+
+/**
+ * @author Patrick Reinhart <patrick@reini.net>
+ */
+@FunctionalInterface
+interface ByteArrayConsumer {
```

Comment:
I am just wondering why you didn't just use a `Consumer(byte[])`

Reply:
This interface is no longer needed get removed with my next commit